icu4c/source/common/uloc_tag.cpp

```diff
@@ -2102,7 +2103,10 @@ ultag_parse(const char* tag, int32_t tagLen, int32_t* parsedLen, UErrorCode* sta
             parsedLenDelta = checkLegacyLen - replacementLen;
             uprv_strcpy(t->buf, LEGACY[i + 1]);
             if (checkLegacyLen != tagLen) {
-                uprv_strcpy(t->buf + replacementLen, tag + checkLegacyLen);
+                uprv_strncpy(t->buf + replacementLen, tag + checkLegacyLen,
+                             oldTagLength - checkLegacyLen);
+                // null terminate after strncpy
```

Comment:
```suggestion
                uprv_memcpy(t->buf + replacementLen, tag + checkLegacyLen,
                            oldTagLength - checkLegacyLen);
                // NUL-terminate after memcpy().
```
When we *know* how many bytes to copy, and never want to even think about a NUL-terminated source string, then we should use memcpy().